debian/sourcemapper.go

```diff
@@ -0,0 +1,170 @@
+package debian
+
+import (
+	"bufio"
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/textproto"
+	"strings"
+	"sync"
+
+	"github.com/quay/zlog"
+	"go.opentelemetry.io/otel/baggage"
+	"go.opentelemetry.io/otel/label"
+)
+
+const sourcesURL = "https://ftp.debian.org/debian/dists/%s/%s/source/Sources.gz"
+
+var (
+	repos = [3]string{"main", "contrib", "non-free"}
+)
+
+type option func(*SourcesMap)
+
+// WithSourcesURL allows callers to override the sourcesURL.
+func WithSourcesURL(url string) option {
+	return func(sm *SourcesMap) {
+		sm.sourcesURL = url
+	}
+}
+
+// WithClient allows callers to override the http.DefaultClient.
+func WithClient(cl *http.Client) option {
+	return func(sm *SourcesMap) {
+		sm.client = cl
+	}
+}
+
+// NewSourcesMap returns a SourcesMap but does not perform any
+// inserts into the map. That needs to be done explitly by calling
+// the Update method.
+func NewSourcesMap(release Release, opts ...option) *SourcesMap {
+	d := &SourcesMap{
+		release:    release,
+		sourcesURL: sourcesURL,
+		mu:         &sync.RWMutex{},
+		sourceMap:  make(map[string]map[string]struct{}),
+		etagMap:    make(map[string]string),
+		client:     http.DefaultClient,
```

Comment:
I'm of the opinion that new code should not be using `DefaultClient` -- if an `http.Client` is needed to work properly it should be an argument to the constructor.